libcontainer/nsenter/nsexec.c

```diff
@@ -52,15 +52,7 @@ enum sync_t {
 /* Stores the current stage of nsexec. */
 int current_stage = STAGE_SETUP;
 
-/* Assume the stack grows down, so arguments should be above it. */
-struct clone_t {
-	/*
-	 * Reserve some space for clone() to locate arguments
-	 * and retcode in this place
-	 */
-	char stack[4096] __attribute__((aligned(16)));
-	char stack_ptr[0];
-
+struct fork_t {
```

Comment:
This struct isn't needed anymore. You can remove `child_func` entirely and make `fork_and_run` look like this:

```c
static int fork_and_run(jmp_buf *env, int jmpval)
{
	pid_t pid = fork();
	if (pid < 0)
		bail("failed to fork");
	if (pid > 0)
		return pid;
	/* Jump back to the big switch in nsexec... */
	longjmp(env, jmpval);
}
```